components/com_fabrik/models/form.php

```diff
@@ -2186,8 +2186,10 @@ public function validate()
 			foreach ($elementModels as $elementModel)
 			{
 				// If the user can't view or edit the element, then don't validate it. Otherwise user sees failed validation but no indication of what failed
-				if (!$elementModel->canUse() && !$elementModel->canView())
-				{
+				// if (!$elementModel->canUse() && !$elementModel->canView())
+				// Bauer note: This is wrong. For example, if !canUse, but canView, this would not be skipped because both are not false! 	
+				// It should be either... if (!$elementModel->canUse() || !$elementModel->canView()) - or...
```

Comment:
I'm not sure of the specific case you are fixing here, but the original case this was for is when the element is not visible on the form - no edit, no view.  So validation failure would not be visible to the user, because no visible element to attach it to.

Reply:
Not sure if they did get your point.

But I have read the code and I think that you may be right that it only makes sense to validate if you can use and view the element. At the moment it validates if either canView or canUse and I also think that, since you need both to update, then you should also need both to validate. (It would not be a good idea to validate any field that user cannot update as the user cannot fix the validation failure.)

Reply:
Please remove the comments and annotate the commit instead.